maxim/tests/test_connection_retry_logic.py

```
@@ -0,0 +1,326 @@
+        ) as mock_session_context:
+            mock_session = Mock()
+            mock_session_context.return_value.__enter__.return_value = mock_session
+
+            # First call fails, second succeeds
+            mock_session.request.side_effect = [connection_error, success_response]
+
+            # Test that get_prompt method works despite connection error
+            try:
+                result = self.maxim_api.get_prompt("test-prompt-id")
+                # If we get here, the retry logic worked
+                self.assertIsNotNone(result)
```

Ruff (0.11.9)

Use a regular `assert` instead of unittest-style `assertIsNotNone`

Replace `assertIsNotNone(...)` with `assert ...`

(PT009)